### ISSUE TITLE
New instance: piped.ducks.party

### ISSUE THREAD

**Issue by nyuuzyou** [TeamPiped/Piped#3414]
### Api Url

https://pipedapi.ducks.party

### Instance location

Netherlands, 🇳🇱

### CDN (not required)

- [ ] My instance does use a CDN.

### Instance owner

- [X] I am the owner of the instance.

### Other details

Provider: Timeweb Cloud
Status/Uptime: [status.ducks.party](https://status.ducks.party)
Registration and logs are disabled.
Piped Proxy: `piped-proxy.ducks.party`
Thanks to all maintainers!

**Comment by Bnyro** [comment]
Added, thanks for hosting an instance!

**Comment by barkoder** [comment]
Your onion instance http://gn72uh2yv5dy5iminsskb22i62ynnhzxbfekcmiw6m6eakw7kuetewid.onion/ is using the clearweb api https://pipedapi.ducks.party . Please have all aspects of the onion instance(fonts api etc) only communicate over onion and not the clearweb. See #922 .

Thanks for running a piped instance.

**Comment by barkoder** [comment]
@nyuuzyou I have no way to contact you except this comment section. ducks.party doesn't have any contact information. So please respond. Thanks.

**Comment by nyuuzyou** [comment]
Hello,
I just want to clarify that the instance is set up using a reverse proxy for onion connections, not a separate container.

To fully utilize onion routing, it would be necessary to run a completely separate stack, as Piped does not currently support dual routing (clearweb/onion). I'm open to the idea of running a separate frontend and specifying the onion address of the backend reverse proxy, but it's important to note that piped-proxy is not specified on the frontend side, but on the backend side.

At the moment, the Piped backend consumes a significant amount of server resources under load. If I launch a second instance, it could negatively impact the performance of all instances hosted by ducks.party.

Right now, there aren't any specific rules for Piped onion instances. I don't think there are any issues at the moment.

**Comment by barkoder** [comment]
Thanks for responding @nyuuzyou

> At the moment, the Piped backend consumes a significant amount of server resources under load. If I launch a second instance, it could negatively impact the performance of all instances hosted by ducks.party.

If this is the case, then that's completely understandable. I'd rather the clearweb users(the majority of users) have performant access to your instance, than someone like me who uses onions.

But also, if this *is* the case, then please consider turning off your onion instance of piped.

Whenever I use an onion instance of piped,

 * piped2bbch4xslbl2ckr6k62q56kon56ffowxaqzy42ai22a4sash3ad.onion

 * piped.g4c3eya4clenolymqbpgwz3q3tawoxw56yhzk4vugqrl6dtu3ejvhjid.onion

 * piped.r4focoma7gu2zdwwcjjad47ysxt634lg73sxmdbkdozanwqslho5ohyd.onion

I use the corresponding onion API/backend to pull vids,

 * pipedihonqyp5fh3ffn5c7i2zlg4lljc4gorefhpydtqiapewy5gprad.onion

 * api.piped.g4c3eya4clenolymqbpgwz3q3tawoxw56yhzk4vugqrl6dtu3ejvhjid.onion

 * pipedapi.r4focoma7gu2zdwwcjjad47ysxt634lg73sxmdbkdozanwqslho5ohyd.onion

The entire purpose of querying over onion is for the additional privacy.

But if the actual videos on an onion instance of piped are being proxied over the clearweb(such as on your onion instance), it defeats privacy benefits of using an onion instance altogether.

The specific problem I have with your onion instance in particular, is that there is no visual UI indicator that shows your onion instance is querying data over the clearweb backend.
So visiting your onion instance makes a regular user *feel* secure, without any privacy benefits of onions.
Basically, your onion instance is not as privacy preserving as the other onion instances I've listed above. But it looks and behaves just like them.
I only figured out that your instance was proxying over the clearweb by opening up devtools(which the majority of people don't do).

Again, if performance of your clearweb instance will be impacted by running a separate onion backend, or if there are other technical challenges, I completely understand.

But until it is feasible that all aspects of your onion instance(including backend) communicate *only* over onions, please consider turning off your onion instance, and removing the `onion-location` header on your clearweb instance.

Thanks.

**Comment by nyuuzyou** [comment]
Hello,

Piped onion instance is now fully operational via Tor, using a unique .onion address rather than a hostname/clearnet.

Thank you for your issue. I'm sorry I couldn't do it earlier due to limited server resources.
